src/classes/routine-service-proxy/routine-service-proxy.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+#include "../service-proxy/service-proxy.hpp"
+#include <memory>
+
+class RoutineServiceProxy : public ServiceProxy {
+public:
+    RoutineServiceProxy(RoutineService& realService, std::map<std::string, ServiceState::state_t> depsMap);
+    ~RoutineServiceProxy() override = default;
+
+protected:
+    friend class ServiceHandler;
+    friend class Tester;
+
+    class RoutineState : public ServiceState {
+
+    public:
+        RoutineServiceProxy& m_upperProxy;
+
+        RoutineState(state_t state, RoutineServiceProxy& upperProxy)
+            : m_upperProxy(upperProxy)
+            , ServiceState(state)
+        {
+        }
```

Comment:
```suggestion
        RoutineState(state_t state, RoutineServiceProxy& upperProxy)
            : m_upperProxy(upperProxy)
            , ServiceState(state) {};
```

Aplicar de forma análoga em todo arquivo

Reply:
Aplicado!!